src/qibo/backends/qulacs.py

```diff
@@ -0,0 +1,81 @@
+import re
+
+import numpy as np
+
+from qibo import __version__
+from qibo.backends import NumpyBackend
+from qibo.result import CircuitResult, QuantumState
+
+
+class QulacsBackend(NumpyBackend):
+
+    def __init__(self):
+        super().__init__()
+        import qulacs  # pylint: disable=import-error
+        from qulacs import (  # pylint: disable=no-name-in-module, import-error
+            QuantumCircuitSimulator,
+            converter,
+        )
+
+        self.qulacs = qulacs
+        self.simulator = QuantumCircuitSimulator
+        self.converter = converter
+        self.name = "qulacs"
+        self.versions = {"qibo": __version__, "qulacs": qulacs.__version__}
+        self.device = "CPU"
+
+    def circuit_to_qulacs(
+        self, circuit: "qibo.Circuit"
+    ) -> "qulacs.QuantumCircuit":  # pylint: disable=no-member
+        """
+        Converts a qibo circuit in a qulacs circuit.
+
+        Args:
+            circuit (:class:`qibo.models.circuit.Circuit`): Input circuit to convert.
+
+        Returns:
+            qulacs.QuantumCircuit: The converted qulacs circuit.
+        """
+        qasm_str = re.sub("^//.+\n", "", circuit.to_qasm())
+        qasm_str = re.sub(r"creg\s.+;", "", qasm_str)
+        qasm_str = re.sub(r"measure\s.+;", "", qasm_str)
+        circ = self.converter.convert_QASM_to_qulacs_circuit(qasm_str.splitlines())
+        return circ
+
+    def execute_circuit(
+        self,
+        circuit: "qibo.Circuit",
+        nshots: int = 1000,
```

Comment:
I think you need to add
```suggestion
        circuit: "qibo.Circuit",
        initial_state=None,
        nshots: int = 1000,
```
here, otherwise executing any circuit using the `GlobalBackend` interface, for example:
```py
import qibo
from qibo import Circuit, gates

qibo.set_backend("qulacs")


c = Circuit(2)
c.add(gates.H(0))
c.add(gates.CNOT(0, 1))

print(c())
```
will lead to
```sh
TypeError: QulacsBackend.execute_circuit() takes from 2 to 3 positional arguments but 4 were
```

It is also weird that this is not captured in the tests, given that this interface is the most basic one, even advertised as the minimum example in our README.

Reply:
Thanks, I'll raise an error in case `initial_state != None` and test it